apps/train_loc/lib/train_loc/manager.ex

```diff
@@ -131,15 +128,71 @@ defmodule TrainLoc.Manager do
     {:noreply, [], state}
   end
 
-  defp update_vehicles(%ManagerEvent{} = manager_event, %{
-         first_message?: first_message?,
-         time_baseline: time_baseline_fn,
-         excluded_vehicles: excluded_vehicles
-       }) do
-    manager_event.vehicles_json
-    |> Enum.reject(&(&1["VehicleID"] in excluded_vehicles))
-    |> vehicles_from_data
+  def handle_new_event(data, state) do
+    json = data["data"]["results"]
+
+    if json do
+      json
+      |> Map.values()
+      |> Enum.map(fn %{
+                       "Heading" => heading,
+                       "Latitude" => latitude,
+                       "Longitude" => longitude,
+                       "Speed" => speed,
+                       "TripID" => trip_id,
+                       "Update Time" => timestamp,
+                       "VehicleID" => vehicle_id
+                     } ->
+        %Vehicle{
+          vehicle_id: vehicle_id,
+          timestamp: timestamp |> DateTime.from_iso8601() |> elem(1),
+          trip: trip_id,
+          latitude: latitude,
+          longitude: longitude,
+          heading: heading,
+          speed: speed
+        }
+      end)
+      |> prune_vehicles(state)
+      |> VehiclePositionsEnhanced.encode()
```

Comment:
@bklebe this is a no-op: you aren't assigning the result of the JSON encoding to a variable. You're then passing the original JSON (as an Elixir map) to `put_object/2` below. That's what's crashing in ECS.

Reply:
thanks for finding this! obviously not what I intended to do.